OpenRA.Mods.D2k/Widgets/Logic/PurchaseWidgetLogic.cs

```diff
@@ -0,0 +1,82 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Mods.Common.Widgets;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.D2k.Widgets.Logic
+{
+	public class PurchseWidgetLogic : ChromeLogic
+	{
+		readonly World world;
+		readonly Lazy<ProductionPaletteWidget> paletteWidget;
+		readonly Lazy<Widget> backgroundBottom;
+		readonly ImageWidget bottomExtension;
+		readonly ButtonWidget purchaseButton;
+		readonly LabelWidget deliveryLabel;
+		readonly LabelWidget timer;
+
+		[ObjectCreator.UseCtor]
+		public PurchseWidgetLogic(Widget widget, World world)
+		{
+			this.world = world;
+			backgroundBottom = Exts.Lazy(() => Ui.Root.GetOrNull("BOTTOM_CAP"));
+			purchaseButton = widget.Get<ButtonWidget>("PURCHASE_BUTTON");
+			bottomExtension = widget.Get<ImageWidget>("BOTTOM_EXTENSION");
+			deliveryLabel = widget.Get<LabelWidget>("DELIVERY_IN");
+			timer = widget.Get<LabelWidget>("DELIVERY_TIMER");
```

Comment:
```suggestion
			deliveryLabel = widget.Get<LabelWidget>("DELIVERY_IN");
			var textCache = new CachedTransform<string, string>(s => TranslationProvider.GetString(DeliveryIn, Translation.Arguments("time", time)));
			deliveryLabel.GetText = () => textCache.Update(time);
```

combine delivery label and counter using localisation